hrjobcontract/CRM/Export/BAO/Export.php

```diff
@@ -0,0 +1,2208 @@
+        $queryMode = CRM_Contact_BAO_Query::MODE_PLEDGE;
+        break;
+
+      case CRM_Export_Form_Select::CASE_EXPORT:
+        $queryMode = CRM_Contact_BAO_Query::MODE_CASE;
+        break;
+
+      case CRM_Export_Form_Select::GRANT_EXPORT:
+        $queryMode = CRM_Contact_BAO_Query::MODE_GRANT;
+        break;
+
+      case CRM_Export_Form_Select::ACTIVITY_EXPORT:
+        $queryMode = CRM_Contact_BAO_Query::MODE_ACTIVITY;
+        break;
+    }
+    return $queryMode;
+  }
+
+  /**
+   * Get default return property for export based on mode
+   *
+   * @param int $exportMode
+   *   Export mode.
+   *
+   * @return string $property
+   *   Default Return property
+   */
+  public static function defaultReturnProperty($exportMode) {
+    // hack to add default return property based on export mode
+    $property = NULL;
+    if ($exportMode == CRM_Export_Form_Select::CONTRIBUTE_EXPORT) {
+      $property = 'contribution_id';
+    }
+    elseif ($exportMode == CRM_Export_Form_Select::EVENT_EXPORT) {
+      $property = 'participant_id';
+    }
+    elseif ($exportMode == CRM_Export_Form_Select::MEMBER_EXPORT) {
+      $property = 'membership_id';
+    }
+    elseif ($exportMode == CRM_Export_Form_Select::PLEDGE_EXPORT) {
+      $property = 'pledge_id';
+    }
+    elseif ($exportMode == CRM_Export_Form_Select::CASE_EXPORT) {
+      $property = 'case_id';
+    }
+    elseif ($exportMode == CRM_Export_Form_Select::GRANT_EXPORT) {
+      $property = 'grant_id';
+    }
+    elseif ($exportMode == CRM_Export_Form_Select::ACTIVITY_EXPORT) {
+      $property = 'activity_id';
+    }
+    return $property;
+  }
+
+  /**
+   * Get Export component
+   *
+   * @param int $exportMode
+   *   Export mode.
+   *
+   * @return string $component
+   *   CiviCRM Export Component
+   */
+  public static function exportComponent($exportMode) {
+    switch ($exportMode) {
+      case CRM_Export_Form_Select::CONTRIBUTE_EXPORT:
+        $component = 'civicrm_contribution';
+        break;
+
+      case CRM_Export_Form_Select::EVENT_EXPORT:
+        $component = 'civicrm_participant';
+        break;
+
+      case CRM_Export_Form_Select::MEMBER_EXPORT:
+        $component = 'civicrm_membership';
+        break;
+
+      case CRM_Export_Form_Select::PLEDGE_EXPORT:
+        $component = 'civicrm_pledge';
+        break;
+
+      case CRM_Export_Form_Select::GRANT_EXPORT:
+        $component = 'civicrm_grant';
+        break;
+    }
+    return $component;
+  }
+
+  /**
+   * Get Query Group By Clause
+   * @param int $exportMode
+   *   Export Mode
+   * @param string $queryMode
+   *   Query Mode
+   * @param array $returnProperties
+   *   Return Properties
+   * @param object $query
+   *   CRM_Contact_BAO_Query
+   *
+   * @return string $groupBy
+   *   Group By Clause
+   */
+  public static function getGroupBy($exportMode, $queryMode, $returnProperties, $query) {
+    // HRJobContract extension override by PCHR-1409.
```

Comment:
there are couples things are needed :

1- overriding the file should be in a commit and the changes happen to this file is in another commit so it will be easier to track down the changes ,

2- add  @custom before the comment to point that this part of the code is a custom code 

```// @custom HRJobContract extension override by PCHR-1409.```

3- document the changes in this page https://compucorp.atlassian.net/wiki/display/PCHR/Overridden+CiviCRM+files